apps/backend/src/drink-action/drink-action.service.ts

```diff
@@ -1,25 +1,44 @@
-import { Injectable, NotFoundException } from '@nestjs/common';
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaClientKnownRequestError } from '@prisma/client/runtime/library';
 import { PrismaService } from 'nestjs-prisma';
 
 import { CreateDrinkActionDto } from './dto/create-drink-action.dto';
+import { DrinkActionDto } from './dto/drink-action.dto';
 import { UpdateDrinkActionDto } from './dto/update-drink-action.dto';
 import { DrinkAction } from './entities/drink-action.entity';
 
 @Injectable()
 export class DrinkActionsService {
   constructor(private prisma: PrismaService) {}
 
-  async create(createDrinkActionDto: CreateDrinkActionDto) {
-    return this.prisma.drinkAction.create({ data: createDrinkActionDto });
+  async create(createDrinkActionDto: CreateDrinkActionDto): Promise<DrinkAction> {
+    try {
+      const { drinkId, eventId, ...rest } = createDrinkActionDto;
+      return await this.prisma.drinkAction.create({
+        data: {
+          ...rest,
+          drink: { connect: { id: drinkId } },
+          event: { connect: { id: eventId } },
+        },
+        include: { drink: true, event: true },
+      });
+    } catch (error) {
+      if (error instanceof PrismaClientKnownRequestError) {
+        if (error.code === 'P2003') {
+          throw new BadRequestException('drinkId or eventId does not exist');
+        }
+      }
+      throw error;
+    }
```

Comment:
When I tested this with invalid event or drink Ids, I got a different Prisma error code, so the response was 500 instead of 400.

Reply:
i dont know how to fix this, can you tell me?

Reply:
if you console.log the error inside the catch block and send a request with an invalid event or drink id, you'll see the error that prisma throws. You can use the code from that instead of P2003